addons/delivery/models/sale_order.py

```diff
@@ -59,6 +59,10 @@ def set_delivery_line(self, carrier, amount):
         self._remove_delivery_line()
         for order in self:
             order.carrier_id = carrier.id
+            if order.state in ('sale', 'done'):
+                pending_deliverys = order.picking_ids.filtered(
+                    lambda p: p.state not in ('done', 'cancel') and not p.move_ids.origin_returned_move_id)
```

Comment:
Shouldn't you use `is_return_picking` instead?

Reply:
Hello, @clesgow  I can't use this, cause when the carrier is set in return, this field is getting true only, and as per the regular flow we have to avoid putting carrier_id for every picking which is return.